app/decorators/service_provider_session_decorator.rb

```diff
@@ -60,4 +72,8 @@ def cancel_link_url
   def request_url
     sp_session[:request_url]
   end
+
+  def openid_connect_redirector
+    OpenidConnectRedirector.from_request_url(request_url)
```

Comment:
There's a DB call doing this each time so we could probably memoize it? Not a priority.